test/models/ui/ranking/ranking_element_details_test.dart

```diff
@@ -83,18 +92,40 @@ void main() {
       final rankingElementDetail1 = RankingElementDetails(
         userDisplayName: testingUserData.displayName,
         userUserName: testingUserData.username,
+        userID: testingUserFirestoreId,
         centauriPoints: 10,
         userRank: 1,
       );
 
       final rankingElementDetail2 = RankingElementDetails(
         userDisplayName: testingUserData.displayName,
         userUserName: testingUserData.username,
+        userID: testingUserFirestoreId,
         centauriPoints: 10,
         userRank: 2,
       );
 
       expect(rankingElementDetail1 == rankingElementDetail2, isFalse);
     });
+
+    test("!= userID", () {
+      final rankingElementDetail1 = RankingElementDetails(
+        userDisplayName: testingUserData.displayName,
+        userUserName: testingUserData.username,
+        userID: testingUserFirestoreId,
+        centauriPoints: 10,
+        userRank: 1,
+      );
+
+      final rankingElementDetail2 = RankingElementDetails(
+        userDisplayName: testingUserData.displayName,
+        userUserName: testingUserData.username,
+        userID: const UserIdFirestore(value: "not_the_same_uid"),
+        centauriPoints: 10,
+        userRank: 1,
+      );
+
+      expect(rankingElementDetail1 == rankingElementDetail2, isFalse);
```

Comment:
If you choose to go with my previous suggestion, I think it could also be applied here.

```suggestion
      expect(rankingElementDetail1, isNot(equals(rankingElementDetail2)));
```

Reply:
You are completely right.

Changed in [cefc851](https://github.com/ProximaEPFL/proxima/pull/348/commits/cefc85186f44b65ca6bdfe9f097aaea631128868)

I also added this matcher to the other tests in this file :)

Thank you !

Reply:
You are right !

Applied in: [c7be8de](https://github.com/ProximaEPFL/proxima/pull/348/commits/c7be8de3f35c0cf986dac822e0e5fe83c1b4c4b1)

Note that I also applied this change for the other tests in the file :)

Thank you !